### DISCUSSION TITLE
Is there a way to optimize  time execution?

### DISCUSSION THREAD

**Discussion by RBouallegue** [kernc/backtesting.py#739] [Ideas]
Thank you for your work in this library.

For huge data frames the backtesting becomes quite slow.

Can we switch to GPU , is there a way to update it to be running using Numba or Cuda.

I am really curious if other people have faced the some problem , and they found workarounds?

**Comment by DominikBerger01** [comment]
When I started working with 1m Data execution time became quite slow. 
For e.g. I have been working with a data set of 118834 rows --> Resampling (done outside of strategy) takes about 7 sec.
Running the backtest takes ~10.5min 

In contrast the same operation with 5m data takes about 10 sec.
I'm not an expert in increasing performance the only thing that I could think of is using Cython in computational heavy parts of the library.

Best regards, 
Dominik
